File: postprocessing/ncx/extern/ezTest-0.0.0/examples/ezTestDemo.cpp
/*
Demo of using ezTestRunner class.
./ezTestDemo --verbose --color

20111107 rsz Created.
*/

#include "ezTest.hpp"

bool test_ezassert(ez::ezTestRunner& runner) {
  ezassert(runner.clean);
  return true;
}
bool test_clean(ez::ezTestRunner& runner) {
  return runner.clean;
}
bool test_tmpdir(ez::ezTestRunner& runner) {
  return runner.tmpdir.compare("/tmp")==0;
}
bool test_verbose(ez::ezTestRunner& runner) {
  return runner.verbose;
}

int main(int argc, const char* argv[]) {
  ez::ezTestRunner t;
  t.getopt(argc, argv);
  t.add(&test_ezassert, "test_EzAssert. See that clean is on by using ezassert test.");
  t.add(&test_clean, "test_Clean. See if --clean is on.");
  t.add(&test_tmpdir, "test_TmpDir. See if --tmpdir is set to \"/tmp\".");
  t.add(&test_verbose, "test_Verbose. See if --verbose was used.");

  // Do setup here and/or in functions.
  
  int status = t.run();
  
  // Do tear down here and/or in functions.
  
  return status;
}

File: postprocessing/ncx/extern/ezTest-0.0.0/examples/ezTestDemo.py
"""
Simple demo of ezTest python interface.

python ezTestDemo.py --color --verbose

2011116 rsz Created.
"""

# Set up path.
import sys
from os.path import normpath, dirname, join
local_module = normpath(join(dirname(__file__), '..'))
sys.path.insert(0, local_module)

from ezTest import *

#######################################################################
class ezTestDemo(ezTestSuite):
  """ A test suite with single setup/teardown and many tests. """
  
  def __init__(self):
    self.name = 'ezTest Example Suite'
    
    # Ordered list of functions to run and descriptive name.
    # The runner will assign a test number.
    # Each test function should return True or False.
    self.tests = [
      [self.testExample1, 'First example test.'],
      [self.testExample2, 'Second example test that fails.'],
    ]
    
  def testExample1(self): return True
  def testExample2(self): return False

#######################################################################
if __name__=='__main__':
  # Create list of suites.
  suites = [ezTestDemo(), ezTestDemo()]
  # Get command-line options.
  cliOptions = ezTestOpt()
  
  runner = ezTestRunner(suites, opt=cliOptions)
  runner.run()